### ISSUE TITLE
Worker node stuck in Cordoned state on a k8s upgrade of the cluster

### ISSUE THREAD

**Issue by sowmyav27** [rancher/rancher#26405]
**What kind of request is this (question/bug/enhancement/feature request):** bug

**Steps to reproduce (least amount of steps as possible):**

- Bring up a cluster with k8s 1.15.11-rancher1-1 in rancher:v2.3.4
- Cluster spec: 1 etcd, 1 control and 3 worker nodes
- Upgrade rancher to v2.4.0
- Upgrade k8s version to 1.17.4-rancher1-3
- Control plane and etcd nodes get upgraded to 1.17 k8s version.
- One worker node gets upgraded.
- 2nd worker node is picked for Upgrade, but is seen stuck in "Cordoned" state (node: `m-62cz7`)
- Logs:
```
2020/03/31 18:18:20 [DEBUG] cluster [c-4djxs] worker-upgrade: workerNodeInfo: nodes 3 maxAllowed 1 upgrading 1 notReady [ m-ccgfr] toProcess [ m-62cz7] toPrepare [ m-v8q4v] done []
2020/03/31 18:18:20 [ERROR] NodeController c-4djxs/m-ccgfr [rke-worker-upgrader] failed with : cluster [c-4djxs] worker-upgrade: more than allowed nodes upgrading for cluster: unavailable 2 maxUnavailable 1
2020/03/31 18:18:20 [INFO] cluster [c-4djxs] worker-upgrade: call upgrade to reconcile for node [m-ccgfr]
2020/03/31 18:18:20 [DEBUG] cluster [c-4djxs] worker-upgrade cluster status node version [2]
2020/03/31 18:18:20 [DEBUG] cluster [c-4djxs] worker-upgrade: node [m-4b9nq] is not a workerOnly node
2020/03/31 18:18:20 [DEBUG] cluster [c-4djxs] worker-upgrade: node [m-vjzvk] is not a workerOnly node
2020/03/31 18:18:20 [DEBUG] cluster [c-4djxs] worker-upgrade: workerNodeInfo: nodes 3 maxAllowed 1 upgrading 1 notReady [] toProcess [ m-62cz7] toPrepare [  m-ccgfr m-v8q4v] done []
2020/03/31 18:18:22 [INFO] cluster [c-4djxs] worker-upgrade: call upgrade to reconcile for node [m-ccgfr]
2020/03/31 18:18:22 [DEBUG] cluster [c-4djxs] worker-upgrade cluster status node version [2]
2020/03/31 18:18:22 [DEBUG] cluster [c-4djxs] worker-upgrade: node [m-4b9nq] is not a workerOnly node
2020/03/31 18:18:22 [DEBUG] cluster [c-4djxs] worker-upgrade: node [m-vjzvk] is not a workerOnly node
2020/03/31 18:18:22 [DEBUG] cluster [c-4djxs] worker-upgrade: workerNodeInfo: nodes 3 maxAllowed 1 upgrading 1 notReady [] toProcess [ m-62cz7] toPrepare [  m-ccgfr m-v8q4v] done []
2020/03/31 18:18:43 [DEBUG] cluster [c-4djxs] worker-upgrade: received node-version [2] for node [m-62cz7]
2020/03/31 18:18:43 [INFO] cluster [c-4djxs] worker-upgrade: updating node [m-62cz7] with node-version 2
2020/03/31 18:18:43 [INFO] cluster [c-4djxs] worker-upgrade: call upgrade to reconcile for node [m-62cz7]
2020/03/31 18:18:43 [DEBUG] cluster [c-4djxs] worker-upgrade cluster status node version [2]
2020/03/31 18:18:43 [DEBUG] cluster [c-4djxs] worker-upgrade: node [m-4b9nq] is not a workerOnly node
2020/03/31 18:18:43 [DEBUG] cluster [c-4djxs] worker-upgrade: node [m-vjzvk] is not a workerOnly node
2020/03/31 18:18:43 [DEBUG] cluster [c-4djxs] worker-upgrade: workerNodeInfo: nodes 3 maxAllowed 1 upgrading 1 notReady [] toProcess [] toPrepare [  m-ccgfr m-v8q4v] done [ m-62cz7]
2020/03/31 18:18:43 [INFO] cluster [c-4djxs] worker-upgrade: sending node-version for node [sowmya-work-1] version 2
2020/03/31 18:18:43 [INFO] cluster [c-4djxs] worker-upgrade: updated node [m-62cz7] to uncordon
2020/03/31 18:18:43 [DEBUG] cluster [c-4djxs] worker-upgrade: return node [m-62cz7], unschedulable field [false]
2020/03/31 18:18:43 [DEBUG] cluster [c-4djxs] worker-upgrade: received node-version [2] for node [m-62cz7]
2020/03/31 18:18:44 [DEBUG] cluster [c-4djxs] worker-upgrade: return node [m-62cz7], unschedulable field [false]
2020/03/31 18:18:44 [INFO] cluster [c-4djxs] worker-upgrade: call upgrade to reconcile for node [m-62cz7]
2020/03/31 18:18:44 [DEBUG] cluster [c-4djxs] worker-upgrade cluster status node version [2]
2020/03/31 18:18:44 [DEBUG] cluster [c-4djxs] worker-upgrade: node [m-4b9nq] is not a workerOnly node
2020/03/31 18:18:44 [DEBUG] cluster [c-4djxs] worker-upgrade: node [m-62cz7] is done
2020/03/31 18:18:44 [DEBUG] cluster [c-4djxs] worker-upgrade: node [m-vjzvk] is not a workerOnly node
2020/03/31 18:18:44 [DEBUG] cluster [c-4djxs] worker-upgrade: workerNodeInfo: nodes 3 maxAllowed 1 upgrading 0 notReady [] toProcess [] toPrepare [  m-ccgfr m-v8q4v] done []
2020/03/31 18:18:44 [INFO] cluster [c-4djxs] worker-upgrade: updated node [m-ccgfr] to cordon
2020/03/31 18:18:44 [DEBUG] cluster [c-4djxs] worker-upgrade: return node [m-ccgfr], unschedulable field [true]
2020/03/31 18:18:44 [INFO] cluster [c-4djxs] worker-upgrade: sending node-version for node [sowmya-work-2] version 2
2020/03/31 18:18:44 [DEBUG] cluster [c-4djxs] worker-upgrade: return node [m-ccgfr], unschedulable field [true]
2020/03/31 18:18:44 [INFO] cluster [c-4djxs] worker-upgrade: call upgrade to reconcile for node [m-ccgfr]
2020/03/31 18:18:44 [DEBUG] cluster [c-4djxs] worker-upgrade cluster status node version [2]
2020/03/31 18:18:44 [DEBUG] cluster [c-4djxs] worker-upgrade: node [m-62cz7] is done
2020/03/31 18:18:44 [DEBUG] cluster [c-4djxs] worker-upgrade: node [m-vjzvk] is not a workerOnly node
2020/03/31 18:18:44 [DEBUG] cluster [c-4djxs] worker-upgrade: node [m-4b9nq] is not a workerOnly node
2020/03/31 18:18:44 [DEBUG] cluster [c-4djxs] worker-upgrade: workerNodeInfo: nodes 3 maxAllowed 1 upgrading 1 notReady [] toProcess [ m-ccgfr] toPrepare [ m-v8q4v] done []
2020/03/31 18:18:45 [INFO] cluster [c-4djxs] worker-upgrade: sending node-version for node [sowmya-work-2] version 2
2020/03/31 18:18:45 [INFO] cluster [c-4djxs] worker-upgrade: call upgrade to reconcile for node [m-62cz7]
2020/03/31 18:18:45 [DEBUG] cluster [c-4djxs] worker-upgrade cluster status node version [2]
2020/03/31 18:18:45 [DEBUG] cluster [c-4djxs] worker-upgrade: node [m-4b9nq] is not a workerOnly node
2020/03/31 18:18:45 [DEBUG] cluster [c-4djxs] worker-upgrade: node [m-62cz7] is done
2020/03/31 18:18:45 [DEBUG] cluster [c-4djxs] worker-upgrade: node [m-vjzvk] is not a workerOnly node
2020/03/31 18:18:45 [DEBUG] cluster [c-4djxs] worker-upgrade: workerNodeInfo: nodes 3 maxAllowed 1 upgrading 1 notReady [] toProcess [ m-ccgfr] toPrepare [ m-v8q4v] done []
2020/03/31 18:18:45 [DEBUG] cluster [c-4djxs] worker-upgrade: received node-version [2] for node [m-ccgfr]
2020/03/31 18:18:45 [INFO] cluster [c-4djxs] worker-upgrade: updating node [m-ccgfr] with node-version 2
2020/03/31 18:18:45 [INFO] cluster [c-4djxs] worker-upgrade: call upgrade to reconcile for node [m-ccgfr]
2020/03/31 18:18:45 [DEBUG] cluster [c-4djxs] worker-upgrade cluster status node version [2]
2020/03/31 18:18:45 [DEBUG] cluster [c-4djxs] worker-upgrade: node [m-4b9nq] is not a workerOnly node
2020/03/31 18:18:45 [DEBUG] cluster [c-4djxs] worker-upgrade: node [m-62cz7] is done
2020/03/31 18:18:45 [DEBUG] cluster [c-4djxs] worker-upgrade: node [m-vjzvk] is not a workerOnly node
2020/03/31 18:18:45 [DEBUG] cluster [c-4djxs] worker-upgrade: workerNodeInfo: nodes 3 maxAllowed 1 upgrading 1 notReady [] toProcess [] toPrepare [ m-v8q4v] done []
2020/03/31 18:18:45 [INFO] cluster [c-4djxs] worker-upgrade: sending node-version for node [sowmya-work-2] version 2
2020/03/31 18:18:46 [DEBUG] cluster [c-4djxs] worker-upgrade: received node-version [2] for node [m-ccgfr]
2020/03/31 18:20:18 [DEBUG] cluster [c-4djxs] worker-upgrade: received node-version [2] for node [m-ccgfr]
2020/03/31 18:20:21 [INFO] cluster [c-4djxs] worker-upgrade: call upgrade to reconcile for node [m-ccgfr]
2020/03/31 18:20:21 [DEBUG] cluster [c-4djxs] worker-upgrade cluster status node version [2]
2020/03/31 18:20:21 [DEBUG] cluster [c-4djxs] worker-upgrade: node [m-ccgfr] is not ready
2020/03/31 18:20:21 [DEBUG] cluster [c-4djxs] worker-upgrade: node [m-4b9nq] is not a workerOnly node
2020/03/31 18:20:21 [DEBUG] cluster [c-4djxs] worker-upgrade: node [m-62cz7] is done
2020/03/31 18:20:21 [DEBUG] cluster [c-4djxs] worker-upgrade: node [m-vjzvk] is not a workerOnly node
2020/03/31 18:20:21 [DEBUG] cluster [c-4djxs] worker-upgrade: workerNodeInfo: nodes 3 maxAllowed 1 upgrading 0 notReady [ m-ccgfr] toProcess [] toPrepare [ m-v8q4v] done []
2020/03/31 18:20:22 [INFO] cluster [c-4djxs] worker-upgrade: call upgrade to reconcile for node [m-ccgfr]
2020/03/31 18:20:22 [DEBUG] cluster [c-4djxs] worker-upgrade cluster status node version [2]
2020/03/31 18:20:22 [DEBUG] cluster [c-4djxs] worker-upgrade: node [m-4b9nq] is not a workerOnly node
2020/03/31 18:20:22 [DEBUG] cluster [c-4djxs] worker-upgrade: node [m-62cz7] is done
2020/03/31 18:20:22 [DEBUG] cluster [c-4djxs] worker-upgrade: node [m-vjzvk] is not a workerOnly node
2020/03/31 18:20:22 [DEBUG] cluster [c-4djxs] worker-upgrade: workerNodeInfo: nodes 3 maxAllowed 1 upgrading 1 notReady [] toProcess [] toPrepare [ m-v8q4v] done []
2020/03/31 18:20:27 [INFO] cluster [c-4djxs] worker-upgrade: call upgrade to reconcile for node [m-ccgfr]
2020/03/31 18:20:27 [DEBUG] cluster [c-4djxs] worker-upgrade cluster status node version [2]
2020/03/31 18:20:27 [DEBUG] cluster [c-4djxs] worker-upgrade: node [m-4b9nq] is not a workerOnly node
2020/03/31 18:20:27 [DEBUG] cluster [c-4djxs] worker-upgrade: node [m-62cz7] is done
2020/03/31 18:20:27 [DEBUG] cluster [c-4djxs] worker-upgrade: node [m-vjzvk] is not a workerOnly node
2020/03/31 18:20:27 [DEBUG] cluster [c-4djxs] worker-upgrade: workerNodeInfo: nodes 3 maxAllowed 1 upgrading 1 notReady [] toProcess [] toPrepare [ m-v8q4v] done []
```

**Expected Result:**
Node should have upgraded and should have Uncordoned, Upgrade on worker nodes should have finished.

**Other details that may be helpful:**
This is not easily reproducible.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.3.4 to 2.4.0
- Installation option (single install/HA): single

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): rke do
- Kubernetes version (use `kubectl version`):

```
1.17.4-rancher1-3
```

gz#13226

**Comment by herrbpl** [comment]
How should one get out from this state? Having similar issue, Windows ( 10.0.17763.1282 ) worker is stuck in upgrade.

**Comment by kinarashah** [comment]
@herrbpl is the node stuck in cordoned before or after the upgrade?

**Comment by arkanmgerges** [comment]
Hi, I have similar issue with cordoned/uncordoned manually I switched from kubectl, and also from Rancher UI. I had problems to run my cluster after a shutdown and starting my vms (openstack) the next day. I used snapshot restore from Rancher UI.
But the state remained for the nodes in "cordoned".
In the following screenshots, look at k8s-worker-2 for example.


![image](https://user-images.githubusercontent.com/7836805/93666282-b1df0980-fa85-11ea-9f43-e4aeadc84459.png)
![image](https://user-images.githubusercontent.com/7836805/93666291-c58a7000-fa85-11ea-97a7-a00c1c9d1028.png)
![image](https://user-images.githubusercontent.com/7836805/93666305-d20ec880-fa85-11ea-8197-d17464b3da4d.png)
